packages/jobs/src/job-definitions/batchEvaluations/runBatchEvaluationJob.ts

```diff
@@ -0,0 +1,88 @@
+import { randomUUID } from 'crypto'
+
+import {
+  Dataset,
+  DocumentVersion,
+  EvaluationDto,
+} from '@latitude-data/core/browser'
+import { findWorkspaceFromDocument } from '@latitude-data/core/data-access'
+import { jobs } from '@latitude-data/core/jobs'
+import { disk } from '@latitude-data/core/lib/disk'
+import { NotFoundError } from '@latitude-data/core/lib/errors'
+import { connection } from '@latitude-data/core/redis'
+import { CommitsRepository } from '@latitude-data/core/repositories'
+import { previewDataset } from '@latitude-data/core/services/datasets/preview'
+import { Job } from 'bullmq'
+
+import { ProgressTracker } from '../../utils/progressTracker'
+
+type RunBatchEvaluationJobParams = {
+  evaluation: EvaluationDto
+  dataset: Dataset
+  document: DocumentVersion
+  runCount?: number
+  offset?: number
+  parametersMap?: Record<string, number>
+  batchId?: string
+}
+
+export const runBatchEvaluationJob = async (
+  job: Job<RunBatchEvaluationJobParams>,
+) => {
+  const {
+    evaluation,
+    dataset,
+    document,
+    offset = 0,
+    parametersMap,
+    runCount: jobRunCount,
+    batchId = randomUUID(),
+  } = job.data
+  const workspace = await findWorkspaceFromDocument(document)
+  if (!workspace) throw new NotFoundError('Workspace not found')
+
+  const commit = await new CommitsRepository(workspace.id)
+    .find(document.commitId)
+    .then((r) => r.unwrap())
+  const fileMetadata = dataset.fileMetadata
+  const runCount = jobRunCount || fileMetadata.rowCount
+  const result = await previewDataset({
```

Comment:
CSV parser has a stream version. I think we should use that version here to avoid loading all the file in memory 
https://csv.js.org/parse/api/stream/